### PR TITLE
Updating ScriptHostOptionsSetup to populate workers configuration. Enabled tests

### PR DESCRIPTION
**Pull request by fabiocav**

### CODE REVIEW

File: sample/host.json [hunk 1]
```diff
@@ -1,33 +1,35 @@
...
     },
-    "functionTimeout": "00:05:00"
+    "functionTimeout": "00:05:00",
+    "logger": {
+        "fileLoggingMode": "always"
+    },
+    "swagger": {
+        "enabled": true
+    },
+    "extensions": {
```
Comment (fabiocav) [inline sample/host.json]: Moving extensions configuration under the `extensions` section.

File: src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs [hunk 2]
```diff
@@ -46,6 +65,69 @@ public void Configure(ScriptJobHostOptions options)
...
+                options.FunctionTimeout = requestedTimeout;
+            }
+            else if (_environment.IsDynamic())
+            {
+                // Apply a default if this is running on Dynamic.
+                options.FunctionTimeout = DefaultFunctionTimeout;
+            }
+
+            // TODO: DI: JobHostOptions need to me updated.
+            //scriptConfig.HostOptions.FunctionTimeout = ScriptHost.CreateTimeoutConfiguration(scriptConfig);
```
Comment (fabiocav) [inline src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs]: Opening this to track:
https://github.com/Azure/azure-functions-host/issues/3256

File: test/WebJobs.Script.Tests.Integration/CosmosDB/CosmosDBEndToEndTestsBase.cs [hunk 3]
```diff
@@ -99,9 +99,6 @@ public override void ConfigureJobHost(IWebJobsBuilder webJobsBuilder)
                     "CosmosDBIn",
                     "CosmosDBOut"
                 };
-
```
Comment (fabiocav) [inline test/WebJobs.Script.Tests.Integration/CosmosDB/CosmosDBEndToEndTestsBase.cs]: removed all of the hardcoded initialization that was peppered throughout the code.

File: test/WebJobs.Script.Tests/Configuration/ScriptHostOptionsSetupTests.cs [hunk 4]
```diff
@@ -102,9 +102,187 @@ public void Configure_AllowPartialHostStartup()
...
+
+            var environment = new TestEnvironment();
+            environment.SetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteSku, "Dynamic");
+
+            var options = GetConfiguredOptions(settings, environment);
+
+            Assert.Equal(ScriptHostOptionsSetup.DefaultFunctionTimeout, options.FunctionTimeout);
+            Assert.Equal(LanguageWorkerConstants.DefaultMaxMessageLengthBytesDynamicSku, options.MaxMessageLengthBytes);
+
+            // TODO: DI Need to ensure JobHostOptions is correctly configured
```
Comment (fabiocav) [inline test/WebJobs.Script.Tests/Configuration/ScriptHostOptionsSetupTests.cs]: Tracked by https://github.com/Azure/azure-functions-host/issues/3256

File: src/WebJobs.Script/Host/ScriptHost.cs [hunk 5]
```diff
@@ -60,13 +60,8 @@ public class ScriptHost : JobHost, IScriptJobHost
         private IPrimaryHostStateProvider _primaryHostStateProvider;
         private string _instanceId;
         // TODO: DI (FACAVAL) Review
```
Comment (mathewc) [inline src/WebJobs.Script/Host/ScriptHost.cs]: remove TODO?

File: test/WebJobs.Script.Tests/Configuration/ScriptHostOptionsSetupTests.cs (hunk 4 above)
Comment (mathewc) [reply test/WebJobs.Script.Tests/Configuration/ScriptHostOptionsSetupTests.cs]: A GA must have, right?

Comment (fabiocav) [reply test/WebJobs.Script.Tests/Configuration/ScriptHostOptionsSetupTests.cs]: Yes, I labeled it as a must.

File: src/WebJobs.Script/Host/ScriptHost.cs (hunk 5 above)
Comment (fabiocav) [reply src/WebJobs.Script/Host/ScriptHost.cs]: 👍

**Pull request merged**